h/views/api_auth.py

```diff
@@ -2,11 +2,100 @@
 
 from __future__ import unicode_literals
 
+from pyramid.httpexceptions import HTTPBadRequest, HTTPFound
+from pyramid.view import view_config, view_defaults
+from pyramid import security
+
 from h.exceptions import OAuthTokenError
 from h.util.view import cors_json_view
 from h.util.datetime import utc_iso8601
 
 
+@view_defaults(route_name='oauth_authorize')
+class OAuthAuthorizeController(object):
+
+    def __init__(self, request):
+        self.request = request
+        self.oauth_svc = self.request.find_service(name='oauth')
+        self.user_svc = self.request.find_service(name='user')
+
+    @view_config(request_method='GET',
+                 renderer='h:templates/oauth/authorize.html.jinja2')
+    def get(self):
+        """
+        Check the user's authentication status and present the authorization
+        page.
+        """
+        self._check_params()
+
+        if self.request.authenticated_userid is None:
+            raise HTTPFound(self.request.route_url('login', _query={
+                              'next': self.request.url}))
+
+        params = self.request.params
+        user = self.user_svc.fetch(self.request.authenticated_userid)
+
+        return {'username': user.username,
+                'client_name': 'Hypothesis',
+                'client_id': params['client_id'],
```

Comment:
Can `client_id` be retrieved from the `authclient` obj that `_check_params()` returns instead?